### PR TITLE
[release/v1.19] Relaxed container runtime version

### LINKED ISSUE #989: PLEG is not healthy - Ubuntu 20.04 - runc issue
In some not deterministic cases, a Nodes will get not ready with the following issue:
```
PLEG is not healthy: pleg was last seen active 11m54.676753691s ago; threshold is 3m0s
```
- Docker/kubelet it-self was still healthy

As @xmudrii  mentioned there is an ref upstream issue: https://github.com/kubernetes/kubernetes/issues/45419

`kubelet` errors of node:
```
Jun 01 16:46:01 node-pool-7fddb5656f-gg9jj kubelet[179303]: E0601 16:46:01.993106  179303 kubelet.go:1799] skipping pod synchronization - [container runtime status check may not have completed yet, PLEG is not healthy: pleg was last seen active 11m54.676753691s ago; threshold is 3m0s]
Jun 01 16:46:05 node-pool-7fddb5656f-gg9jj kubelet[179303]: E0601 16:46:05.825510  179303 remote_runtime.go:140] StopPodSandbox "dd4c40eb9b25ae13bdf11ac76457853cff15fbfcad653d4c7aab73ec24df6ac4" from runtime service failed: rpc error: code = DeadlineExceeded desc = context deadline exceeded
Jun 01 16:46:05 node-pool-7fddb5656f-gg9jj kubelet[179303]: E0601 16:46:05.825576  179303 kuberuntime_gc.go:176] Failed to stop sandbox "dd4c40eb9b25ae13bdf11ac76457853cff15fbfcad653d4c7aab73ec24df6ac4" before removing: rpc error: code = DeadlineExceeded desc = context deadline exceeded
Jun 01 16:46:06 node-pool-7fddb5656f-gg9jj kubelet[179303]: E0601 16:46:06.993297  179303 kubelet.go:1799] skipping pod synchronization - [container runtime status check may not have completed yet, PLEG is not healthy: pleg was last seen active 11m59.676925345s ago; threshold is 3m0s]
Jun 01 16:46:07 node-pool-7fddb5656f-gg9jj kubelet[179303]: E0601 16:46:07.386039  179303 kuberuntime_manager.go:951] PodSandboxStatus of sandbox "ca3b771d7176f8f3899d9fa2b510beb2db7bd2f95a4260e7bb3ab89a4b7fd80b" for pod "oidcconfigcontroller-service-5587f56f57-hwnpc_infra-security(e696a633-409e-47ed-bb9d-91f7c78445f6)" error: rpc error: code = DeadlineExceeded desc = context deadline exceeded
Jun 01 16:46:11 node-pool-7fddb5656f-gg9jj kubelet[179303]: E0601 16:46:11.993579  179303 kubelet.go:1799] skipping pod synchronization - [container runtime status check may not have completed yet, PLEG is not healthy: pleg was last seen active 12m4.677175232s ago; threshold is 3m0s]
Jun 01 16:46:16 node-pool-7fddb5656f-gg9jj kubelet[179303]: E0601 16:46:16.993864  179303 kubelet.go:1799] skipping pod synchronization - [container runtime status check may not have completed yet, PLEG is not healthy: pleg was last seen active 12m9.677504093s ago; threshold is 3m0s]
```

## Envrionment:
- KKP `2.17.0`, machine-controller: `v.1.27.1`
- Ubuntu image itself is used from march with `20.04.02` without `distupgrade=false`, Kernel `5.4.0-70`
- Kubernetes `1.19.8`
- containerd verison `1.4.3-2`
- docker `19.03.13~3-0`

## Reproducable
As mentioned, it could be reproduced https://github.com/kubernetes/kubernetes/issues/45419#issuecomment-810385662
Will be tested by @mfahlandt if upgrade of versions to 
- `1.19.11`
- latest `1.20.x` could somehow solve the issue

## Next Steps:
- Investigation if the downgroad is a long-term fix for the machine-controller
- If not, find another fix
- Fix issue in machine controller and rollout a new KKP bugfix version

## Workaround for manual fix
As suggested at https://github.com/kubernetes/kubernetes/issues/45419#issuecomment-827038034, downgrade containerd.io to the `1.4.3-1*` version what holds runc in version `1.0.0-rc92` fixed the issue.

```
ssh root@10.53.11.23 -- apt install containerd.io=1.4.3-1*

# check version
ssh root@10.53.11.23 -- runc -version
```

An automated script for all nodes:
**ATTENTION:** could cause restarts of potential all pods in your cluster!
```bash
#!/bin/bash
cd $(dirname $(realpath $0))
FOLDER=$(pwd)
user='root'

set -euo pipefail
kubectl get nodes --no-headers -o wide | awk '{print $7}' > $FOLDER/hosts.txt
cat $FOLDER/hosts.txt

### exclude '#' lines
grep -v '#' $FOLDER/hosts.txt| while read -r host; do
  echo "$user@$host"
  ssh "$user@$host" -oStrictHostKeyChecking=no -t 'bash -s' <<EOF
echo '------------------'
runc -version
echo '>>>>> apt install containerd.io=1.4.3-1*'
apt-get install --allow-downgrades --allow-change-held-packages -y containerd.io=1.4.3-1*
runc -version
EOF
done

```

### PR DESCRIPTION
**Pull request by kron4eg**
**What this PR does / why we need it**:
To update to the latest runc that features security issues and fixes #989

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Backport fix for #989

**Special notes for your reviewer**:

**Optional Release Note**:

```release-note
Relaxed container runtime version
```

### CODE REVIEW

File: .prow.yaml [hunk 1]
```diff
@@ -217,6 +217,28 @@ presubmits:
               memory: 1Gi
               cpu: 500m
 
+  - name: pull-machine-controller-e2e-aws-arm
```
Comment (xmudrii) [inline .prow.yaml:220]: ARM support has been added in [v1.21.0](https://github.com/kubermatic/machine-controller/releases/tag/v1.21.0). Does it make sense to test ARM machines with an unsupported version of MC?

Comment (kron4eg) [reply .prow.yaml:220]: No, that's just a relic of cherrypick from far future :D 
Also there is some RHEL fix is missing and it keep failing :(

File: pkg/userdata/rhel/provider.go [hunk 2]
```diff
@@ -194,13 +194,20 @@ write_files:
...
+        More info at: https://bugzilla.redhat.com/show_bug.cgi?id=1756473 */}}
     sed -i 's/\$releasever/7/g' /etc/yum.repos.d/docker-ce.repo
     yum-config-manager --save --setopt=docker-ce-stable.module_hotfixes=true
 
     DOCKER_VERSION='{{ .DockerVersion }}'
-    yum install -y docker-ce-${DOCKER_VERSION} \
-      docker-ce-cli-${DOCKER_VERSION} \
+    yum install -y docker-ce-${DOCKER_VERSION} docker-ce-cli-${DOCKER_VERSION}
+
+    {{- if eq .CloudProviderName "azure" }}
```
Comment (moadqassem) [inline pkg/userdata/rhel/provider.go:204]: The current place where you add this line is kinda useless as the line number 194: `yum install -y yum-utils ` is already calling yum with the default rhui packages. Please move this block from this line to line number 193.

**Review by moadqassem — CHANGES_REQUESTED** [review]

File: pkg/userdata/rhel/provider.go (hunk 2 above)
Comment (kron4eg) [reply pkg/userdata/rhel/provider.go]: thanks!

**Review by moadqassem — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**